Source/JavaScriptCore/parser/Parser.cpp

```diff
@@ -5090,7 +5105,24 @@ template <class TreeBuilder> TreeExpression Parser<LexerType>::parsePrimaryExpre
         if (UNLIKELY(match(ARROWFUNCTION)))
             return 0;
 
-        return createResolveAndUseVariable(context, ident, *ident == m_vm.propertyNames->eval, start, location);
+        // If encounter JavaScript code `arguments.length`, a fast path can be applied to get
+        // `arguments.length` directly from CallFrame without materializing `arguments` object
+        // first. However, we should avoid that when parsing array fuctnion or other property accesses.
```

Comment:
I don't think this is the right place to insert these code. Why not adding these kind of things into parseMemberExpression as I said?